core/arch/aarch64/decode.c

```diff
@@ -67,8 +67,41 @@ byte *
 decode_eflags_usage(dcontext_t *dcontext, byte *pc, uint *usage,
                     dr_opnd_query_flags_t flags)
 {
-    *usage = 0; /* FIXME i#1569 */
-    return pc + 4;
+    uint res = 0;
+    instr_t instr;
+    int opc;
+    /* XXX: We should perhaps add flag information in codec.txt rather than list
+     * all the opcodes here, although the list is short and unlikely to change.
```

Comment:
Yeah this was going to be my comment that you would expect this to be part of the central ISA "database" and not separate, and I think it has to be because this routine decode_eflags_usage() is not normally used: decode() is expected to fill in eflags information.  decode() is supposed to be level 3, and eflags is at level 2 (decode_opcode() is supposed to give you opcode + eflags, but not operands; decode() also gives operands).

I guess instr_get_eflags() calls decode_eflags_usage() so it happens to work, but the intent there is that we have not fully decoded the instruction yet if we don't have flags info: it assumes it's level 1 where all we did was figure out the size (which takes some effort on x86) and thus we need to do a decode from raw bytes to get the flags info.  But for A64, there's no reason to call decode_common() twice: the flags info should be there from the first call.